newdle/client/src/util/timezones.js

```diff
@@ -0,0 +1,43 @@
+import preval from 'preval.macro';
+import 'moment-timezone';
+
+// This will just generate a pre-compiled list with all timezones and their
+// respective offsets.
+export const commonTimezones = preval`
+  require('moment-timezone');
+  const moment = require('moment');
+  const _ = require('lodash');
+
+  const tzData = require('moment-timezone/data/meta/latest.json');
+  const commonTzSet = Object.entries(tzData.countries).reduce((accum, [key, {zones}]) => {
+    zones.forEach(tz => {
+      accum[tz] = true;
+    })
+    return accum;
+  }, {});
+
+  // add some common timezones which are not official in any country
+  [
+    // UTC/GMT
+    'UTC',
+    'GMT',
+    // USA
+    'US/Eastern', 'US/Pacific', 'US/Mountain',
+    'US/Central', 'US/Arizona', 'US/Hawaii', 'US/Alaska',
+    // Canada
+    'Canada/Newfoundland', 'Canada/Atlantic', 'Canada/Eastern',
+    'Canada/Central', 'Canada/Mountain', 'Canada/Pacific'
+  ].forEach(k => {
+    commonTzSet[k] = true;
+  });
+
+  const timezones = Object.keys(commonTzSet).map(k => {
+    const z = moment.tz.zone(k);
+    return {
+      name: z.name,
+      offset: z.utcOffset(+ new Date()),
+      caption: moment.tz(k).format('z (Z)')
```

Comment:
the zone caption looks pretty strange for some timezones:

![image](https://user-images.githubusercontent.com/179599/65241637-5382f200-dae4-11e9-9f04-0d1edbf9f643.png)

Reply:
What do you mean?

Reply:
`+14 (+14:00)` looks pretty strange to me

Reply:
Ah yes. Maybe I can remove the first part when it's just numbers/signs.